designer/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala

```diff
@@ -88,68 +85,29 @@ class DBProcessService(deploymentService: DeploymentService,
   import cats.instances.future._
   import cats.syntax.either._
 
-  override def getProcessState(processIdWithName: ProcessIdWithName)(implicit user: LoggedUser, ec: ExecutionContext): Future[ProcessState] =
-    deploymentService.getProcessState(processIdWithName)
-
   override def archiveProcess(processIdWithName: ProcessIdWithName)(implicit user: LoggedUser): Future[EmptyResponse] =
     withNotArchivedProcess(processIdWithName, ProcessActionType.Archive) { process =>
       if (process.isSubprocess) {
         archiveSubprocess(process)
       } else {
+        // FIXME: This doesn't work correctly because concurrent request can change a state and double archive actions will be done.
+        //        See ManagementResourcesConcurrentSpec and how DeploymentService handles it correctly for deploy and cancel
         doOnProcessStateVerification(process, ProcessActionType.Archive)(doArchive)
       }
     }
 
   override def unArchiveProcess(processIdWithName: ProcessIdWithName)(implicit user: LoggedUser): Future[EmptyResponse] =
     withProcess(processIdWithName) { process =>
       if (process.isArchived) {
-        repositoryManager.runInTransaction(
+        repositoryManager.runInTransaction(DBIOAction.seq(
```

Comment:
Should we have also `run(Seq[DBIOAction])`, runAsTry(DBIOAction]), runFlatten(DBIOAction)? etc. If slick provides such nice combinators, we should use it instead of writting a new that just take two combinators and join them. I know that for someone who better know our code than slick's API it may be a little bit easier to have everything on our side. But I think that we should invest in spreading knowledge about slick's API than our code. Slick is used in many places, Nu is only one. TBH IMO this class should has only one `run()` method, even `runTransactionally` is too much. Ask someone with slick's knowledge what do our `RepositoryManager.runTranactionally` and after that ask what do `DBIOAction.transactionally`